src/adaptors/tcp_adaptor.c

```diff
@@ -1132,18 +1140,24 @@ static qdr_tcp_connection_t *qdr_tcp_connection_egress(qd_tcp_connector_t *conne
         qd_log(tcp_adaptor->log_source, QD_LOG_INFO,
                "[C%"PRIu64"] call pn_proactor_raw_connect(). Egress connecting to: %s",
                tc->conn_id, tc->bridge->host_port);
+
+        LOCK(tc->activation_lock);
```

Comment:
same here: although no log call, pn_raw_connection() does a lot of work as well - stuff like allocating memory. In general we should try to minimize the duration of holding a lock to only the exact critical region.